fs-storage/src/file_storage.rs

```diff
@@ -120,47 +188,26 @@ impl FileStorage {
         log::info!(
             "{} {} entries have been written",
             self.label,
-            value_by_id.len()
+            value_map.len()
         );
         Ok(())
     }
+}
 
-    /// Remove file at stored path
-    pub fn erase(&self) -> Result<()> {
-        fs::remove_file(&self.path).map_err(|err| {
-            ArklibError::Storage(self.label.clone(), err.to_string())
-        })
-    }
-
-    /// Verify the version stored in the file header
-    fn verify_version(&self, header: &str) -> Result<()> {
-        if !header.starts_with(STORAGE_VERSION_PREFIX) {
-            return Err(ArklibError::Storage(
-                self.label.clone(),
-                "Unknown storage version prefix".to_owned(),
-            ));
-        }
-
-        let version = header[STORAGE_VERSION_PREFIX.len()..]
-            .parse::<i32>()
-            .map_err(|_err| {
-                ArklibError::Storage(
-                    self.label.clone(),
-                    "Failed to parse storage version".to_owned(),
-                )
-            })?;
-
-        if version != STORAGE_VERSION {
-            return Err(ArklibError::Storage(
-                self.label.clone(),
-                format!(
-                    "Storage version mismatch: expected {}, found {}",
-                    STORAGE_VERSION, version
-                ),
-            ));
-        }
-
-        Ok(())
+impl<K, V> AsRef<BTreeMap<K, V>> for FileStorage<K, V>
+where
+    K: FromStr
```

Comment:
Do you need all these traits for this implementation? It's best to define only the trait constraints that are necessary.

Reply:
done